src/dex/balancer-v1/balancer-v1.ts

```diff
@@ -377,15 +377,20 @@ export class BalancerV1EventPool {
   }
 
   getPoolPrices(
+    from: Token,
+    to: Token,
     pool: WeightedPool,
     poolData: any,
     side: SwapSide,
     amount: bigint,
   ) {
     if (
       side === SwapSide.BUY &&
-      amount * 2n > BigInt(poolData.balanceOut.toFixed(0))
+      amount * 2n >
+        BigInt(poolData.balanceOut.toString().slice(0, -to.decimals))
     ) {
+      console.log(poolData.balanceOut.toString(), amount * 2n);
+      console.log(BI_MAX_INT);
```

Comment:
I don't understand these changes.  `amount` is already in the token decimals so why are you removing digits?  BigNumber from bignumber.js `toString()` is not good to use because it gives exponential notation if the number is big, use `toFixed(0)` whenever you're converting integers from that library.  Console logs shouldn't be here, and probably in this condition returning `0n` is preferred to indicate no price.